### DISCUSSION TITLE
Thermostat keeps getting turned of seemingly randomly

### DISCUSSION THREAD

**Discussion by LennardPlay** [jmcollin78/versatile_thermostat#317] [Q&A]
I noticed that one on my thermostats turns off randomly when VT is installed, but I can't tell why. Security state stays secure and window detection is stating "window closed"
![grafik](https://github.com/jmcollin78/versatile_thermostat/assets/60149904/a7e65f8c-2749-4e95-acea-64077d416267)
![grafik](https://github.com/jmcollin78/versatile_thermostat/assets/60149904/1c9814a5-4bac-46ff-9e13-986dbf3cba92)

If it helps, these are the attributes of the malfunctioning thermostat:
`hvac_modes:
  - "off"
  - heat
min_temp: 7
max_temp: 35
preset_modes:
  - none
  - frost
  - eco
  - comfort
  - boost
current_temperature: 19.4
temperature: 21
hvac_action: heating
preset_mode: comfort
is_on: false
hvac_mode: "off"
type: null
is_controlled_by_central_mode: true
last_central_mode: null
frost_temp: 10
eco_temp: 18
boost_temp: 23
comfort_temp: 21
frost_away_temp: 10
eco_away_temp: 17
boost_away_temp: 20
comfort_away_temp: 19
power_temp: 13
target_temperature_step: null
ext_current_temperature: -1.4
ac_mode: false
current_power: null
current_power_max: null
saved_preset_mode: comfort
saved_target_temp: 21
saved_hvac_mode: null
window_state: "off"
motion_state: null
overpowering_state: null
presence_state: null
window_auto_state: "off"
window_bypass_state: false
security_delay_min: 60
security_min_on_percent: 0.5
security_default_on_percent: 0.1
last_temperature_datetime: "2024-01-06T10:52:11.465401+01:00"
last_ext_temperature_datetime: "2024-01-06T10:52:11.465401+01:00"
security_state: false
minimal_activation_delay_sec: 10
device_power: 1
mean_cycle_power: null
total_energy: 0.05
last_update_datetime: "2024-01-06T11:13:05.050809+01:00"
timezone: Europe/Berlin
window_sensor_entity_id: binary_sensor.wohnzimmer_fenster_offnet
window_delay_sec: 0
window_auto_enabled: true
window_auto_open_threshold: 3
window_auto_close_threshold: 0
window_auto_max_duration: 30
motion_sensor_entity_id: null
presence_sensor_entity_id: null
power_sensor_entity_id: null
max_power_sensor_entity_id: null
temperature_unit: °C
is_device_active: false
ema_temp: 19.42
is_over_climate: true
start_hvac_action_date: null
underlying_climate_0: climate.wohnzimmer_thermostat
underlying_climate_1: null
underlying_climate_2: null
underlying_climate_3: null
is_regulated: true
regulated_target_temperature: 22.5
auto_regulation_mode: auto_regulation_light
regulation_accumulated_error: 9.899999999999991
auto_fan_mode: auto_fan_none
current_auto_fan_mode: auto_fan_none
auto_activated_fan_mode: null
auto_deactivated_fan_mode: null
friendly_name: Wohnzimmer Heizung
supported_features: 17
`

I would appreciate if someone would assist me with troubleshooting

**Comment by LennardPlay** [comment]
Is this just normal behavior? It seems like VT is turning off my radiator thermostats when the right temperature is reached.
Can I turn this off and have my thermostats always on (expect when manually disabling)

**Comment by jmcollin78** [comment]
Hello @LennardPlay ,

No this is not a normal behavior of VTherm. VTherm never turns off a heater only for window open detection.

Because you have an over_climate VTherm type, it is llikely the underlying which turns off (and VTherm follow). Look at your `climate.wohnzimmer_thermostat` if there is a configuration or something like that which turns it off.

If you to be sure, you will find some logs (in INFO leve) that say's something like "underlying have change its state to off"

**Reply by LennardPlay** [reply]
I get this warning in my logs even if my climate entity is turned on.
My guess is that my climate entity has the state `heat` instead of `on`

`Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: custom_components.versatile_thermostat.underlyings
Source: custom_components/versatile_thermostat/underlyings.py:143
Integration: Versatile Thermostat (documentation, issues)
First occurred: 10:01:59 (1 occurrences)
Last logged: 10:01:59

VersatileThermostat-Schlafzimmer-climate.schlafzimmer_thermostat - The hvac mode is heat, but the underlying device is not ON. Turning on device climate.schlafzimmer_thermostat if needed`

**Reply by jmcollin78** [reply]
This log is totally normal at startup. It indicates a gap between VTherm state and underlying device and it will resynchronize the both. This is not turning off the underluying but on the contrary it turns it on (if needed as said in the message).

What you have to look for is message like "underlying have change its state to off" (I don't have the exact message here, sorry)

**Reply by LennardPlay** [reply]
Alright, I will keep on searching for clues

**Reply by jmcollin78** [reply]
When I have the exact message I will send it here.

**Comment by 0verEngineer** [comment]
Hey, it seems that i have the same problem, i checked the logs of the thermostat itself:
![thermostat_log](https://github.com/user-attachments/assets/6c307b5c-d2a4-46e2-9088-fde109975015)
It says it was changed to off at 12:38:56 PM

And here is the log of the VTherm:
![vtherm_log](https://github.com/user-attachments/assets/3c76b5ed-e9a9-43ea-8756-c7b481bd162e)

It says it was changed to off at: 12:38:55 PM which is one second before the normal climate themostat is turned off

I never turned them off manually and no automation is in place that turns them off.

**Comment by jmcollin78** [comment]
Hello all,

for all this case, you should look at your logs and look for a message like this:
```
VersatileThermostat-Clim salon - Underlying climate climate.clim_salon_tuyalocal have changed. new_hvac_mode is off (vs heat), ....
```

This means that your VTherm have been turned off because the underlying climate have been turned off. Maybe by a manual operation on its remote command, or maybe by the device itself due to internal rules.

This is very common issues. Once again, VTHerm will NEVER turns on or off a device in `over_climate` mode but only when window is detected as open if you have configured like that.

To reproduce the message above, I simply turn off my AC device with the remote command while VTherm was on.

**Reply by 0verEngineer** [reply]
1. As you can see in my logs, VTherm is turned off first which makes no sense because you say VTherm does never turn off the climate thermostat
2. Where do i have to look for the log message you provided?

**Reply by jmcollin78** [reply]
In the home-assistant.log file which is accessible through Samba or VScode addon.
